src/Symfony/Component/Form/Extension/Core/DataTransformer/MoneyToLocalizedStringTransformer.php

```diff
@@ -69,7 +69,17 @@ public function reverseTransform(mixed $value): int|float|null
         $value = parent::reverseTransform($value);
         if (null !== $value && 1 !== $this->divisor) {
             $value = (string) ($value * $this->divisor);
-            $value = 'integer' === $this->modelType ? (int) $value : (float) $value;
+
+            if ('float' === $this->modelType) {
+                return (float) $value;
+            }
+
+            $value = (int) $value;
```

Comment:
We need to do the comparison before doing the cast (see https://3v4l.org/4uaZL).